src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerExecutor.cs

```diff
@@ -51,6 +52,7 @@ internal class AnalyzerExecutor
         private readonly CompilationAnalysisValueProviderFactory _compilationAnalysisValueProviderFactory;
         private readonly CancellationToken _cancellationToken;
 
+        private ConcurrentDictionary<SyntaxTree, SemanticModel> _lazySemanticModelMap;
```

Comment:
Why was this added?

Reply:
This was discussed at the initial design meeting. Almost all our analyzer action callbacks expose a shared semantic model, and large number of bad perf analyzers that we have seen are the ones that fetch their own semantic model for semantic queries. It was an explicit design decision to expose a shared semantic model for all suppressors.